src/CRM/CivixBundle/Resources/views/Code/entity-dao.php.php

```diff
@@ -0,0 +1,18 @@
+<?php
+echo "<?php\n";
+$_namespace = preg_replace(':/:', '_', $namespace);
+?>
+use <?php echo $_namespace ?>_ExtensionUtil as E;
+
+/**
+ * DAOs provide an OOP-style facade for reading and writing database records.
+ *
+ * DAOs are a primary source for metadata in several versions of CiviCRM (<5.75)
+ * and are required for some subsystems (such as APIv3).
+ *
+ * This stub provides compatibility. It is not intended to be modified in a
+ * substantive way. However, you may add comments and annotations.
+ */
+class <?php echo $daoClassName ?> extends CRM_Core_DAO {
```

Comment:
Unfortunately this file probably won't do anything but cause errors. Since it doesn't extend `CRM_Core_DAO_Base` it's kinda like implementing an interface without all the required methods (CRM_Core_DAO doesn't have a *real* interface, but there are certain functions that are expected to exist).

Reply:
Oh, right. We'd talked about the need for a backport of `CRM_Core_DAO_Base`. Haven't gotten that far yet...

Reply:
(Don't want to think about that backport yet -- should probably move toward testing `master`/ `CRM_Core_DAO_Base` in the interim.)

Reply:
OK, as of a5ecaad7cde99c59 (using `CRM_Core_BAO_Base`), I think a bunch of random issues should be addressed. Tests pass for all suites involving `5.74` and `master`. Then, as you'd expect, it fails for everything older - because we don't have the substitute for `CRM_Core_BAO_Base` yet.

So that's the next on the queue... and then circle-back to manually test some real-world upgrades... and then (*knock on wood*) maybe that's it...

Reply:
@colemanw So this is a version of `CRM_Core_DAO_Base` that seems run on older versions (https://github.com/civicrm/civicrm-core/commit/1bcfda4044b46119a8ac92fc841cf35da6ada82f) -- at least, I did light testing with core@5.51, `cv en MYEXT`,  `cv api4 MyEntity.get`, and `cv api4 MyEntity.create`.

I've been going back-forth about how to wire that implementation into `civix` templates -- or, more specifically, whether it's better to:

1. Lean on "reloadable" conventions for DAO-base (*i.e. always using the dynamic class*). 
    * https://github.com/totten/civix/commit/b122378469b8fe20c30e8594ffca80fdcea3701b
2. Lean more on "static" conventions for DAO-base (*i.e. for users on 5.74+, the aliases will be static and visible to the IDE; only use the magic-base for <5.74*). 
     * https://github.com/totten/civix/commit/352a2b48d589bc817a284c42bd94d8ac4829b867

But I would like to get some test-results, so I'm gonna go ahead and push one of them onto this branch.

Reply:
OK, it's using the more static-looking convention (`CRM_Myext_DAO_Base` presents as the base-class; internally, its mapped to either `CRM_Core_DAO_Base` or `CiviMix\Schema\MyExt\DAO`.) It took a few more revisions on other topics, but that arrangement does seem to be passing.